### ISSUE TITLE
Vertebra missegmentation in Gold Standard CT dataset

### ISSUE THREAD

**Issue by Smir-Nik** [wasserth/TotalSegmentator#394]
Hello Dr. Wasserthal,

My name is Nikolai Smirnov, I'm a radiologist.
I recently conducted a review of the data from the TotalSegmentator CT set. To be precise, I randomly selected about 100 cases from your set of Gold Standards downloaded from [here](https://zenodo.org/records/10047292). 
During the review I considered only the segmentation of the vertebrae and noticed that some of the cases (about 10-15% in my set) have a well-marked external contour, but the vertebrae are mixed together.
Perhaps these inaccuracies are the reason that the TotalSegmentator itself periodically mixes the vertebrae as well.
I am attaching several screenshots and patient numbers that you can check to see the errors.

![0350](https://github.com/user-attachments/assets/292001ba-3f71-4acc-90e0-6a7d28ee7086)
![0071](https://github.com/user-attachments/assets/7cea1e0f-edcd-44a3-996a-0802b7be41db)

S0941
S0350
S0513
S0603
S0676
S0071
S0128

I will be glad to help if you need help with reviewing these segmentation.

Best regards,
[Nikolai Smirnov](www.linkedin.com/in/nikolai-smirnov-3b7576234)

**Comment by lassoan** [comment]
I had a look at s0941. The "ground truth" segmentation indeed has errors in separation of 3 vertebrae. Segmentation result mostly good, but there is still an error in 1 vertebra.

<img width="760" alt="image" src="https://github.com/user-attachments/assets/84114fc0-d298-4b7a-a214-1430f51f1dfb">

**Comment by wasserth** [comment]
There are indeed several cases with errors in the vertebrae ground truth in the training data (the same is true for the ribs). Fixing these kind of errors is very time consuming. On our agenda it has low priority. Therefore it did not happen so far. If anybody wants to correct these errors and submit fixed ground truth files I would be very happy to retrain the model.

**Comment by lassoan** [comment]
> Unfortunately, I'm familiar and fast in 3DSlicer, but not very good at writing scripts, and exporting it all separately to NIFTI manually takes a very long time.

This should be no problem at all. ChatGPT should be able to write all the scripts that are needed for this, but I can help with that, too, if you get to that point.

**Comment by gokceay** [comment]
@Smir-Nik If you need any help, I can assist with correcting the segmentations. I have been using Slicer and TotalSegmentator for my projects for a while, and it would be ideal to use high-quality segmentations to train TotalSegmentator. I was also lookong ways to improve the totalsegmentator model. 

@lassoan If we provide TotalSegmentator with more data, its segmentation predictions will improve, right? Also, is it possible to increase the resolution of TotalSegmentator by reducing the slice distance from 1.5 to 0.5?

**Comment by wasserth** [comment]
If providing TotalSegmentator with more and better data, then the segmentations will improve. But it is not guaranteed by how much they will improve.
TotalSegmentator is based on 1.5mm resolution. This is a good tradeoff between accuracy and speed and memory requirements. At the moment I am not planning to switch to 0.5mm resolution. This result in a lot higher runtime and memory requirements.

https://annotate.totalsegmentator.com/ is a very new website I created for people to help with the annotations. So this is now the best way if you want to contribute to the totalsegmentator dataset.

**Comment by fedorov** [comment]
@wasserth I am not experienced with your new website (and I am not a radiologist to contribute any of the corrections for the annotations), but can you confirm - are you saying that the so called "Gold Standards" images with incorrect segmentations of vertebrae and ribs is already included in that website, and are prioritized for review by its users?

It seems that correcting the known problems with the "Gold Standards" data should be absolutely top priority.

For your reference, we applied TotalSegmentator to segment >120,000 lung screening CT scans, and we see invalid vertebrae/ribs segmentations even in the first handful of random images picked from that large cohort.

**Comment by Smir-Nik** [comment]
Dear dr @wasserth and @lassoan.
